### PR TITLE
Pin sphinx version (fix for current docs build errors)

### PR DESCRIPTION
**Pull request by awaelchli**
# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?   
- [ ] Did you write any new necessary tests?  
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CHANGELOG.md)?

## What does this PR do?
Fixes docs build error currently seen on master and in #1379, #1377
Was also suggested by @jeremyjordan 

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: docs/requirements.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-sphinx>=2.0
+sphinx>=2.0,<3.0
```
Comment (Borda) [inline docs/requirements.txt:1]:
```suggestion
sphinx>=2.0, <3.0
```

**Comment by awaelchli** [comment]
@Borda looks like github action is running sphinx 2.2, so 2.4 is too high?

File: docs/source/conf.py [hunk 2]
```diff
@@ -66,7 +66,7 @@
 
 # If your documentation needs a minimal Sphinx version, state it here.
 
-needs_sphinx = '1.4'
+needs_sphinx = '2.4'
```
Comment (awaelchli) [inline docs/source/conf.py:69]:
```suggestion
needs_sphinx = '2.0'
```

Comment (Borda) [reply docs/source/conf.py:69]: then 2.2 :]

**Pull request merged**